lib/cosmos/utilities/message_log.rb

```diff
@@ -39,34 +40,38 @@ def initialize(tool_name, log_dir = nil)
     #
     # @param message [String] Message to write to the log
     def write(message)
-      if @file.nil? or @file.closed? or (not File.exist?(@filename))
-        start()
-      end
+      @mutex.synchronize do
+        if @file.nil? or @file.closed? or (not File.exist?(@filename))
+          start(false)
+        end
 
-      @file.write(message)
-      # While it's nice to flush the IO this is an extreme slowdown
-      #@file.flush
+        @file.write(message)
+      end
     end
 
     # Closes the message log and marks it read only
-    def stop
+    def stop(take_mutex = true)
+      @mutex.lock if take_mutex
       if @file and not @file.closed?
         @file.close
         Cosmos.set_working_dir do
           File.chmod(0444, @filename)
         end
       end
+      @mutex.unlock if take_mutex
     end
 
     # Creates a new message log and sets the filename
-    def start
+    def start(take_mutex = true)
```

Comment:
I see why stop has the take_mutex option because it's used within start. Any reason start has this option? Wouldn't it always need to take the mutex?